ipaclient/install/ipa_client_automount.py

```diff
@@ -177,18 +177,27 @@ def configure_xml(fstore):
         print("Configured %s" % authconf)
 
 
-def configure_nsswitch(fstore, options):
+def configure_nsswitch(statestore, options):
     """
-    Point automount to ldap in nsswitch.conf. This function is for non-SSSD
-    setups only
+    Point automount to ldap in nsswitch.conf.
+    This function is for non-SSSD setups only.
     """
-    fstore.backup_file(paths.NSSWITCH_CONF)
-
     conf = ipachangeconf.IPAChangeConf("IPA Installer")
     conf.setOptionAssignment(':')
 
-    nss_value = ' files ldap'
+    # https://pagure.io/freeipa/issue/8038
+    with open(paths.NSSWITCH_CONF, 'r') as f:
+        current_opts = conf.parse(f)
+        current_nss_value = conf.findOpts(
+            current_opts, name='automount', type='option'
+        )[1]['value']
+        statestore.backup_state(
+            'ipa-client-automount-nsswitch', 'previous-automount',
+            current_nss_value
+        )
+        f.close()
```

Comment:
You don't need f.close(). When the context goes out of scope it will close the file.

Reply:
Thanks. I failed to remove it when I converted to "with open() as f"...